sample/android/src/main/java/co/yml/ychat/android/presentation/completions/CompletionsViewModel.kt

```diff
@@ -0,0 +1,64 @@
+package co.yml.ychat.android.presentation.completions
+
+import androidx.compose.runtime.mutableStateListOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import co.yml.ychat.YChat
+import co.yml.ychat.android.ui.components.output.OutputBoxState
+import kotlinx.coroutines.launch
+
+internal class CompletionsViewModel(private val yChat: YChat): ViewModel() {
+
+    val message = mutableStateOf("")
+
+    val outputBoxStates = mutableStateListOf<OutputBoxState>()
+
+    val onEnableButton = mutableStateOf(message.value.isNotEmpty())
+
+    val onEnableTextField = mutableStateOf(true)
+
+    fun requestCompletions() = viewModelScope.launch {
+        val messageToSend = message.value
+        outputBoxStates.clear()
+        outputBoxStates.add(OutputBoxState.Text(messageToSend))
+        onLoading(true)
+        onMessage("")
+        val completions = yChat.completion()
+            .setMaxTokens(MAX_TOKENS)
+            .setInput(messageToSend)
+        runCatching { completions.execute() }
+            .also { onLoading(false) }
```

Comment:
I think this piece of code will not be executed when there's a failure and will keep the loading being displayed/

Reply:
I see, interesting!